src/compute-client/src/service.rs

```diff
@@ -163,6 +164,16 @@ pub struct PartitionedComputeState<T> {
     /// the tracking state maintained for it and b) we won't re-initialize tracking for a subscribe
     /// we have already dropped.
     pending_subscribes: BTreeMap<GlobalId, PendingSubscribe<T>>,
+    /// Pending responses for a copy to; returnable once all are available.
+    ///
+    /// Tracking of responses for a COPY TO is initialized when the first `CopyResponse` for that command
+    /// is received. Once all shards have provided a `CopyResponse`, a unified copy response is
+    /// emitted and the copy_to tracking state is dropped again.
+    ///
+    /// The compute protocol requires that exactly one response is emitted for each COPY TO command. This
+    /// property ensures that a) we can eventually drop the tracking state maintained for a copy
+    /// and b) we won't re-initialize tracking for a copy we have already served.
+    pending_copy_tos: BTreeMap<GlobalId, BTreeMap<usize, CopyToResponse>>,
```

Comment:
```suggestion
    copy_to_responses: BTreeMap<GlobalId, BTreeMap<usize, CopyToResponse>>,
```
For consistency with `peek_responses`.

Reply:
Also, a nit: Can we move this field after `peek_responses`? And also move the `CopyToResponse` handling below after the `PeekResponse` handling? That will make it easier to see that these are similar, and easier to avoid that they become inconsistent with each other.